### ISSUE TITLE
Command message_agent with string key throws error

### ISSUE THREAD

**Issue by lolxdmainkaisemaanlu** [Significant-Gravitas/Auto-GPT#390]
-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-=
SYSTEM:  Command read_file returned: - Update social media profiles - Collect and analyze market data - Research competitors - Arrange business meetings - Design marketing material
ENTREPRENEUR-GPT THOUGHTS: Let's choose a task to assign to the agent.
REASONING: Given that we have a list of tasks from the saved file, it makes sense to choose one of them, since we are currently running autonomously with no user input.
PLAN:
-  Choose a simple task from the previously generated list to assign to the agent.
CRITICISM: We need to be careful in choosing the task for the agent, as it should align with our goals and be within the scope of the agent's abilities.
NEXT ACTION:  COMMAND = message_agent ARGUMENTS = {'key': 'task_agent', 'message': 'Please research competitors and provide a short report on their strengths and weaknesses.'}
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for Entrepreneur-GPT...
Input:y
-=-=-=-=-=-=-= COMMAND AUTHORISED BY USER -=-=-=-=-=-=-=
SYSTEM:  Command message_agent returned: Error: invalid literal for int() with base 10: 'task_agent'
ENTREPRENEUR-GPT THOUGHTS: Let's create a new agent to work on the task of researching competitors. We'll use GPT-3.5 to generate a short prompt for the agent to complete this task. Before we do that, we need to start the agent.
REASONING: Given that we have a task for the agent, it makes sense to start one and assign the task to it.
PLAN:
-  Start a new agent using the 'start_agent' command.
-  Use GPT-3.5 to generate a prompt for the agent that specifies the task of researching competitors.
-  Assign the task to the new agent.
CRITICISM: We need to ensure that the prompt generated by GPT-3.5 is clear and concise, so that the agent is able to complete the task without any issues.
NEXT ACTION:  COMMAND = start_agent ARGUMENTS = {'name': 'competitor_research_agent', 'task': 'Research competitors and provide a short report on their strengths and weaknesses', 'prompt': 'Please research our
 competitors and provide a brief report on their strengths and weaknesses. Please include information on what sets each competitor apart from our business, and any areas where we could improve to stay ahead of
 the competition.'}
Enter 'y' to authorise command, 'y -N' to run N continuous commands, 'n' to exit program, or enter feedback for Entrepreneur-GPT...

**Comment by onekum** [comment]
I'm getting a similar error but it's `invalid literal for int() with base 10: 'gpt_agent-1'`.

**Comment by mpuels** [comment]
The issue is that in `commands.py::message_agent()` we allow `key` to be of type `str`:

https://github.com/Torantulino/Auto-GPT/blob/7e108d8ab37d828c3d7b129399d8d9927154f801/scripts/commands.py#L266-L267

But then in `agent_manager.py::message_agent()` we actually only accept a `key` of type `int`:

https://github.com/Torantulino/Auto-GPT/blob/7e108d8ab37d828c3d7b129399d8d9927154f801/scripts/agent_manager.py#L34-L37

But since I'm not familiar with the code-base I'm not sure what the best way is to fix this issue.

**Comment by JakFourie** [comment]
Getting the same error

**Comment by meadow-kun** [comment]
same error

**Comment by james431987** [comment]
I often get invalid literal for int() with base 10. 

I wish it would just always use a number and keep track of what the numbers are for

**Comment by vfilinkov** [comment]
Same thing when COMMAND = message_agent

```
NEXT ACTION:  COMMAND = message_agent ARGUMENTS = {'key': 'agent1', 'message': 'Download COVID-19 dataset from https://storage.googleapis.com/covid19-open-data/v3/location-2021-04-05.csv.gz'}
SYSTEM:  Command message_agent returned: Error: invalid literal for int() with base 10: 'agent1'
```

**Comment by zarex5** [comment]
Same error here
_SYSTEM:  Command message_agent returned: Error: invalid literal for int() with base 10: 'Meme Agent'_

**Comment by vibrant** [comment]
Same issue here

**Comment by Qoyyuum** [comment]
message_agent function have improved. Please pull latest changes from master repo and test again. @lolxdmainkaisemaanlu

**Comment by Pwuts** [comment]
Fixed in #1507
